### PR TITLE
TDEE data input

### PR DESCRIPTION
**Pull request by mattpauls**
This is a looping data input to record the date, weight, and calories of an entry. I tried to make it as easy as possible to enter dates by auto-incrementing the date on subsequent entries by a day and by using `dateutil` to parse the date for me.

I'm not super stoked with the logic to prompt the user to enter weight and calories, I feel like there should be a way to use Rich's [FloatPrompt](https://rich.readthedocs.io/en/stable/reference/prompt.html#rich.prompt.FloatPrompt) or [IntPrompt](https://rich.readthedocs.io/en/stable/reference/prompt.html#rich.prompt.FloatPrompt) to clean things up a lot, but I'm not sure how to still have the ability to exit with `e` if those type check for int/float, respectively. Reading the docs, I get a feeling that it might be possible but I have no idea how to go about implementing that.

### CODE REVIEW

File: tdee_macro_calc/main.py [hunk 1]
```diff
@@ -1,12 +1,19 @@
+from curses import panel
 import json
+from datetime import datetime, date, timedelta
+from operator import add
+from xmlrpc.client import Boolean
```
Comment (opensourcedev79) [inline tdee_macro_calc/main.py:5]: Use the `Boolean` type from Python's standard library `typing` for type annotations.

```
from typing import Boolean
```

File: tdee_macro_calc/main.py [hunk 2]
```diff
@@ -1,12 +1,19 @@
+from curses import panel
 import json
+from datetime import datetime, date, timedelta
+from operator import add
```
Comment (opensourcedev79) [inline tdee_macro_calc/main.py:4]: I don't think this import is used anywhere?

File: tdee_macro_calc/main.py [hunk 3]
```diff
@@ -1,12 +1,19 @@
+from curses import panel
```
Comment (opensourcedev79) [inline tdee_macro_calc/main.py:1]: I don't think this import is use anywhere?

**Comment by opensourcedev79** [comment]
I opened #7 against this branch to propose one way we could remove some duplicate logic in this PR. Take a look and let me know what you think! 

If we end up merging #7, it will be added as part of this PR (branch of a branch 😎).

File: tdee_macro_calc/main.py (hunk 1 above)
Comment (mattpauls) [reply tdee_macro_calc/main.py:5]: Good catch. I noticed a few imports that weren't necessary but forgot to go back and clean them up. I'm not sure how that one in particular snuck in there, VS Code must've thrown it in when I wasn't looking somehow.

**Comment by mattpauls** [comment]
@opensourcedev79 I'm pretty happy with where things are at with this, for the moment at least. I'm sure there's plenty to fine-tune but I'm ready to merge and move on from this for now, since it's functional. Thoughts?

**Review by opensourcedev79 — APPROVED** [review]

**Comment by opensourcedev79** [comment]
Looks good! Merging!

**Pull request merged**